src/analysis/normed_space/normed_group_hom.lean

```diff
@@ -458,6 +461,26 @@ def range : add_subgroup V₂ := f.to_add_monoid_hom.range
 lemma mem_range (v : V₂) : v ∈ f.range ↔ ∃ w, f w = v :=
 by { rw [range, add_monoid_hom.mem_range], refl }
 
+lemma comp_range : (g.comp f).range = add_subgroup.map g.to_add_monoid_hom f.range :=
+begin
+  erw add_monoid_hom.map_range,
+  refl,
+end
+
+lemma mem_comp_range (x : V₁) : g (f x) ∈ (g.comp f).range :=
```

Comment:
Isn't `range` defeq to `{ x // ... }`, so that the proof becomes `\<x, rfl\>`?
I think this lemma is a bit specialized, and I guess that usually `simp only [mem_range, comp_apply]` will get you where you want to be.